File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int beautySum(string s) {
        int count = 0;

        int n = s.length();

        for(int i=0;i<n;i++)
        {

            int arr[26] = {0};
            for(int j=i;j<n;j++)
            {
                arr[s[j]-'a']++;
                int mini = INT_MAX;
                int maxi = INT_MIN;

                for(auto it: arr)
                {
                    if(it > 0)
                    {
                        maxi = max(maxi, it);
                        mini = min(mini, it);
                    }
                }
                count+= maxi-mini;
            }
        }
        return count;
        
    }
};

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        
        int maxcount = 0;
        int n = s.length();

        for(int i=0;i<n;i++)
        {
            int[] freq = new int[26];
            for(int j=i;j<n;j++)
            {
                int maxi = Integer.MIN_VALUE;
                int mini = Integer.MAX_VALUE;

                int ch = s.charAt(j)- 'a';
                freq[ch]++;

                for(int it: freq)
                {
                    if(it > 0)
                    {
                        maxi = Math.max(maxi, it);
                        mini = Math.min(mini, it);
                    }
                }
                maxcount += maxi - mini;
            }
        }
        return maxcount;
    }
}